playback/src/audio_backend/alsa.rs

```diff
@@ -228,54 +236,96 @@ impl Sink for AlsaSink {
 
 impl SinkAsBytes for AlsaSink {
     fn write_bytes(&mut self, data: &[u8]) -> io::Result<()> {
-        let mut processed_data = 0;
-        while processed_data < data.len() {
-            let data_to_buffer = min(
-                self.period_buffer.capacity() - self.period_buffer.len(),
-                data.len() - processed_data,
-            );
-            self.period_buffer
-                .extend_from_slice(&data[processed_data..processed_data + data_to_buffer]);
-            processed_data += data_to_buffer;
-            if self.period_buffer.len() == self.period_buffer.capacity() {
-                self.write_buf()?;
-                self.period_buffer.clear();
-            }
-        }
-
+        self.process_bytes(data, 0)?;
```

Comment:
If this is the only place you're calling `process_bytes`, why shell it out to a separate function instead of having the lines of code here?

Reply:
I'm actually going to replace that with a rust `loop` statement. I will put the `loop` where the while loop was. In my testing `loop` is the fastest way to chew though the bytes. Of the 3 (while loop, recursive function and loop) surprisingly the while loop is the slowest.

Reply:
Done.